### ISSUE TITLE
feat: Toggle the visibility of the `ion-menu` in the `ion-split-pane`

### ISSUE THREAD

**Issue by AlexRMU** [ionic-team/ionic-framework#28566]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already include this feature request, without success.

### Describe the Feature Request

Let's say we have such a structure:
```html
<ion-split-pane when="xs" content-id="main">
    <ion-menu content-id="main">
        <ion-header>
            <ion-toolbar>
                <ion-buttons slot="start">
                    <ion-menu-button></ion-menu-button>
                </ion-buttons>
                <ion-title>Menu</ion-title>
            </ion-toolbar>
        </ion-header>
        <ion-content>Menu Content</ion-content>
    </ion-menu>

    <div class="ion-page" id="main">
        <ion-header>
            <ion-toolbar>
                <ion-title>Main View</ion-title>
            </ion-toolbar>
        </ion-header>
        <ion-content>Main View Content</ion-content>
    </div>
</ion-split-pane>
```
I want there to be such an option or opportunity:
The `ion-menu` is hidden on the small screen. When you click on the `ion-menu-button`, the `ion-menu` appears and disappears as a separate `ion-menu`.
On the big screen, the `ion-menu` is shown in the `ion-split-pane`, the `ion-menu-button` is not hidden. When you click on the `ion-menu-button`, the `ion-menu` is hidden and shown as a separate `ion-menu`, but already in the `ion-split-pane`.

At the moment, this can be done, but using some hacking.

### Describe the Use Case

It is strange that this has not been done before.
The `ion-menu` should be able to hide and show. The `ion-split-pane` simply displays the menu on the side on the big screen, where it can also be hidden or shown, and on the small screen it hides, but it can be opened as usual.

**Comment by liamdebeasi** [comment]
Thanks for the report. I'm not sure I understand what's being requested here. When the menu is displayed in a split pane, it is designed to show up next to the main content. If you want the menu to continue to overlay your application then you should not be using split pane.

**Comment by AlexRMU** [comment]
I'll try to explain it in another way.

If the screen matches the "when" media query:
- The `ion-split-pane` is displayed as it is now.
- Additionally, if there is an `ion-menu-button`, when you click on it, the part of the `ion-split-pane` that contains the `ion-menu` appears/disappears.

Else:
- The part of the `ion-split-pane` that contains the `ion-menu` disappears, as it is now.
- Additionally, if there is an `ion-menu-button`, when you click on it, a part of the `ion-split-pane` containing the `ion-menu` appears in the form of a regular `ion-menu`.

In short, you need to add the functionality of a regular `ion-menu` to the `ion-menu` in the `ion-split-pane`.

**Comment by liamdebeasi** [comment]
> Additionally, if there is an ion-menu-button, when you click on it, the part of the ion-split-pane that contains the ion-menu appears/disappears.

Why do you want this behavior? The purpose of the `when` property is to control when the menu is displayed inline with the main content. If the `ion-menu-button` hides the menu that defeats the purpose of the `when` property.

**Comment by AlexRMU** [comment]
Then you can create another property, such as a `breakpoint`.
I would like this `ion-menu` to not just disappear from the `ion-split-pane`, as it is now, but to be able to appear as a separate `ion-menu`, unless of course there is a `ion-menu-button`, so that you don't have to create the `ion-menu` a second time and manually.

**Comment by liamdebeasi** [comment]
Do you have a demo that shows the current behavior as well as the behavior you want to achieve? Ideally a video or a runnable sample.

**Comment by AlexRMU** [comment]
Code:
```html
<ion-split-pane when="md" content-id="main">
    <ion-menu side="end" content-id="main">
        <ion-header>
            <ion-toolbar>
                <ion-title>Menu</ion-title>
            </ion-toolbar>
        </ion-header>
        <ion-content>Menu Content</ion-content>
    </ion-menu>

    <div class="ion-page" id="main">
        <ion-header>
            <ion-toolbar>
                <ion-title>Main View</ion-title>
                <ion-buttons slot="end">
                    <ion-menu-button auto-hide="false"></ion-menu-button>
                    <!-- auto-hide="false" !!! -->
                </ion-buttons>
            </ion-toolbar>
        </ion-header>
        <ion-content>Main View Content (min-width: 768px)</ion-content>
    </div>
</ion-split-pane>
```
Current behavior:

https://github.com/ionic-team/ionic-framework/assets/61969794/f0b3f55c-2b9d-4523-ac7b-cba2fed73e23

Desired behavior:

https://github.com/ionic-team/ionic-framework/assets/61969794/8e5562a3-aed9-4964-80d1-d112282af65d

**Comment by liamdebeasi** [comment]
Thanks for the added information. I recommend using the following approach to build this behavior: https://codepen.io/liamdebeasi/pen/mdvjwwz

Here's a video of the CodePen in action:

https://github.com/ionic-team/ionic-framework/assets/2721089/6021e349-e8f4-454f-a826-462348b45b2f

At the moment we don't have plans to build this feature into Ionic since it can be achieved using the current APIs.
